src/bcs/serializable/moveStructs.ts

```diff
@@ -68,9 +68,10 @@ export class MoveVector<T extends Serializable & EntryFunctionArgument>
    */
   serializeForScriptFunction(serializer: Serializer): void {
     // runtime check to ensure that you can't serialize anything other than vector<u8>
-    // TODO: consider adding support for MoveString later?
     const isU8 = this.values[0] instanceof U8;
-    if (!isU8) {
+    // if the inner array is length 0, we can't check the type because it has no instance, so we assume it's a u8
+    // it may not be, but we don't care because it's serialized as `0x0` regardless
+    if (!isU8 && this.values[0] !== undefined) {
```

Comment:
Actually, sometimes 0 length is serialized as `0x` with no value in the API.  Unsure why

Reply:
ooh yes, you're actually right, I'll say `either way it's serialized to a single byte value: 0`

It's `0x` because they're indicating that it's a byte vector that's just being represented as a string. There's no values after it because it's a 0-length array (hence the serialized `0`)